.github/workflows/ci-core.yml

```diff
@@ -244,6 +244,93 @@ jobs:
             echo "path_output=${resultsFile}" >> $GITHUB_OUTPUT
           fi
 
+  detect-flakey-tests:
+    needs: [filter, core]
+    name: Flakey Test Detection
+    runs-on: ubuntu-latest
+    if: ${{ always() && github.actor != 'dependabot[bot]' }}
+    env:
+      CL_DATABASE_URL: postgresql://postgres:postgres@localhost:5432/chainlink_test?sslmode=disable
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v4.2.1
+      - name: Setup node
+        if: ${{ needs.filter.outputs.changes == 'true' }}
```

Comment:
maybe we could move it up one level from steps to job? since all but one steps need it anyway. And then keep `if` condition only for `Replace chainlink-evm deps` step?